src/lib/FreeC/Backend/Agda/Syntax.hs

```diff
@@ -3,7 +3,59 @@
 
 module FreeC.Backend.Agda.Syntax
   ( module Agda.Syntax.Concrete
+  , module Agda.Syntax.Common
+  , module Agda.Syntax.Position
+  , name
+  , intLiteral
+  , simpleImport
+  , lambda
+  , qname
+  , qname'
   )
 where
 
 import           Agda.Syntax.Concrete
+import           Agda.Syntax.Position
+import           Agda.Syntax.Common
+
+import qualified Agda.Syntax.Common            as Agda
+import qualified Agda.Syntax.Concrete          as Agda
+import qualified Agda.Syntax.Literal           as Agda
+import qualified Agda.Syntax.Position          as Agda
+
+-- | Create a qualified name given a local variable 'Name' and a list of module
+--   'Name's.
+qname :: Agda.Name -> [Agda.Name] -> QName
+qname n = foldr Agda.Qual (Agda.QName n)
+
+-- | Creates a qualified name using an empty list of module names.
+qname' :: Agda.Name -> Agda.QName
+qname' = flip qname []
+
+-- | Creates an @open import@ declaration for the given qualified name.
+--
+--   @open import [modName]@
+simpleImport :: Agda.QName -> Agda.Declaration
+simpleImport modName = Agda.Import
+  Agda.NoRange
+  modName
+  Nothing
+  Agda.DoOpen
+  (Agda.ImportDirective Agda.NoRange Agda.UseEverything [] [] Nothing)
+
+-- | Creates a (not qualified) Agda variable name from a 'String'.
+name :: String -> Agda.Name
+name ident = Agda.Name Agda.NoRange Agda.InScope [Agda.Id ident]
```

Comment:
I would move this to the other smart constructors for names and maybe add some sections like the following.

```haskell
-------------------------------------------------------------------------------
-- Names                                                                     --
-------------------------------------------------------------------------------

…

-------------------------------------------------------------------------------
-- Imports                                                                   --
-------------------------------------------------------------------------------

…

-------------------------------------------------------------------------------
-- Expressions                                                               --
-------------------------------------------------------------------------------

…
```